apps/server/src/modules/tool/common/uc/tool-permissions.uc.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { AuthorizableReferenceType, AuthorizationContext, AuthorizationService } from '../../../authorization';
+import { ContextExternalTool } from '../../context-external-tool/domain';
+import { ContextTypeMapper } from '../mapper';
+
+@Injectable()
+export class ToolPermissionsUc {
+	constructor(private readonly authorizationService: AuthorizationService) {}
+
+	public async ensureContextPermissions(
+		userId: EntityId,
+		contextExternalTool: ContextExternalTool,
+		context: AuthorizationContext
+	): Promise<void> {
+		if (contextExternalTool.id) {
+			await this.authorizationService.checkPermissionByReferences(
```

Comment:
Please use authorizationService.checkPermission and use the https://github.com/hpi-schul-cloud/schulcloud-server/blob/cac5dc5bd3a04965b3a3ab7faf8d67c0ce91b355/apps/server/src/modules/authorization/authorization-context.builder.ts#L4-L4